module3/src/MyList.java

```diff
@@ -0,0 +1,697 @@
+import java.util.*;
+
+/**
+ * Created by yurii.pyvovarenko on 11/28/2021.
+ */
+public class MyList<T> implements List<T>, Iterable<T> {
+    private static final Object[] DEFAULTCAPACITY_EMPTY_ELEMENTDATA = {};
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int GROW_FACTOR = 1;
+
+    transient Object[] elementData;
+    private int size = 0;
+    private int capacity = DEFAULT_CAPACITY;
+
+    MyList() {
+        this.elementData = DEFAULTCAPACITY_EMPTY_ELEMENTDATA;
+    }
+
+    /**
+     * Returns the number of elements in this list.  If this list contains
+     * more than <tt>Integer.MAX_VALUE</tt> elements, returns
+     * <tt>Integer.MAX_VALUE</tt>.
+     *
+     * @return the number of elements in this list
+     */
+    @Override
+    public int size() {
+        return size;
+    }
+
+    /**
+     * Returns <tt>true</tt> if this list contains no elements.
+     *
+     * @return <tt>true</tt> if this list contains no elements
+     */
+    @Override
+    public boolean isEmpty() {
+        if (size < 1) {
```

Comment:
can be simplified to `return size < 1`. Do you understand, why?

Reply:
or even to `size == 0`

Reply:
since size cannot be negative

Reply:
You are correct.
`return size < 1;` will return result or comparison `size < 1`